app/src/main/java/com/example/wishlist/adapters/ProductAdapter.kt

```diff
@@ -33,11 +35,15 @@ class ProductAdapter(val productList: List<Product>, val listener: OnProductClic
                 listener.onProductClicked(product.id)
             }
         }
+    }
+}
 
-
+private class ProductDiffCallback: DiffUtil.ItemCallback<Product>(){
+    override fun areItemsTheSame(oldItem: Product, newItem: Product): Boolean {
+        return oldItem.id == newItem.id
     }
 
-    override fun getItemCount(): Int {
-        return productList.size
+    override fun areContentsTheSame(oldItem: Product, newItem: Product): Boolean {
+        return oldItem == newItem
```

Comment:
Does this check all attributes of the product?

Reply:
Tested evaluation and it does check all attributes.